components/apimgt/org.wso2.carbon.apimgt.gateway/src/main/java/org/wso2/carbon/apimgt/gateway/handlers/security/SchemaValidator.java

```diff
@@ -102,14 +102,12 @@ public boolean handleRequest(MessageContext messageContext) {
             JSONObject payloadObject = getMessageContent(messageContext);
             if (!APIConstants.SupportedHTTPVerbs.GET.name().equals(requestMethod) &&
                     payloadObject != null && !APIMgtGatewayConstants.EMPTY_ARRAY.equals(payloadObject)) {
-                try {
-                    validateRequest(messageContext);
-                } catch (APIManagementException e) {
-                    logger.error("Error occurred while validating the API request", e);
-                }
+                validateRequest(messageContext);
             }
         } catch (IOException | XMLStreamException e) {
             logger.error("Error occurred while building the API request", e);
```

Comment:
SInce we just logging the error if there is any exception will validation get ommitted?

Reply:
Yes, I fixed the issue by setting return false.